### PR TITLE
Add task for bowtie aligner

### PR DESCRIPTION
**Pull request by johnmous**

### CODE REVIEW

File: bowtie.wdl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    File read1
+    String sam
+    String? precommand
+    Int? threads
+    String? otherOptions
+
+    command {
+        ## https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/
+        set -eo pipefail
+        mkdir -p
```
Comment (DavyCats) [inline bowtie.wdl]: No directory name is given here

File: bowtie.wdl [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        bowtie \
+        ${otherOptions}
+        ${"--threads" + threads} \
+        ${reference} \
+        ${read1} > \
+        ${sam}
+    }
+}
+
+task Bowtie_pe {
```
Comment (DavyCats) [inline bowtie.wdl]: Isn't it possible to make a generic bowtie task for both paired and single end?

File: bowtie.wdl [hunk 3]
```diff
@@ -0,0 +1,43 @@
+task Bowtie_se {
```
Comment (rhpvorderman) [inline bowtie.wdl]: `bowtie.wdl` will be imported as `bowtie` by default. Which means these tasks are called as `bowtie.Bowtie_se`.
I propose that you change this to just `task se {` the task can then be called as `bowtie.se`

File: bowtie.wdl (hunk 2 above)
Comment (rhpvorderman) [inline bowtie.wdl]: Bowtie_pe -> pe

File: bowtie.wdl (hunk 1 above)
Comment (rhpvorderman) [inline bowtie.wdl]: Which dir are you creating here?
I always use `mkdir -p $(dirname ${sam})` to make sure any file creation attempts with the `>` sign work.

File: bowtie.wdl [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    File reference
+    File read1
+    String sam
+    String? precommand
+    Int? threads
+    String? otherOptions
```
Comment (rhpvorderman) [inline bowtie.wdl]: I disagree with other options.
If other options are required, another pull request should be made. Or alternatively, the entire command should be written out (which will take a lot of time, so may not be the best option for now).

File: bowtie.wdl (hunk 2 above)
Comment (johnmous) [reply bowtie.wdl]: I could give it a try

File: bowtie.wdl [hunk 5]
```diff
@@ -0,0 +1,43 @@
+task Bowtie_se {
+    File reference
+    File read1
+    String sam
+    String? precommand
+    Int? threads
```
Comment (rhpvorderman) [inline bowtie.wdl]: You added threads here, but did not include a runtime section that uses the threads.
```WDL
runtime {
  cpu: ${default=1 threads}
}
```
You can also set the default here
```WDL
  Int? threads = 1
```
Which looks cleaner. But I have had issues with overriding this default from the config file. I should file a bug for that.

File: bowtie.wdl [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    String? otherOptions
+
+    command {
+        ## https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/
+        set -eo pipefail
+        mkdir -p
+        ${precommand}
+        bowtie \
+        ${otherOptions}
```
Comment (rhpvorderman) [inline bowtie.wdl]: you forgot a \ here. Also otherOptions should preferably be removed

File: bowtie.wdl [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+task Bowtie_pe {
+    File reference
+    File read1
+    File read2
+    String sam
+    String? precommand
+    Int? threads
+    String? otherOptions
```
Comment (rhpvorderman) [inline bowtie.wdl]: See comments above

File: bowtie.wdl [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+    command {
+        set -eo pipefail
+        mkdir -p
```
Comment (rhpvorderman) [inline bowtie.wdl]: see comments above

File: bowtie.wdl [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        ${precommand}
+        bowtie \
+        ${"--threads" + threads} \
+        ${reference} \
+        -1 ${read1} |
```
Comment (rhpvorderman) [inline bowtie.wdl]: | -> \

**Pull request closed without merging**